### PR TITLE
feat(protocol-designer): use RWS for deck setup with highlight and dnd

### PR DESCRIPTION
**Pull request by b-cooper**
Add liquid state and step/substep hover highlighting of relevant labware and wells to the new v2
labware equipped deck setup. Recreate labware drag and drop to move into new deck setup component as well.

## Review Requests

- create a protocol in PD
- add v2 tipracks (not from the labware selection modal yet, only the file data page) and labware (from the labware selection modal)
- move labware to new locations in deck setup step
- create transfer steps and ensure liquid and hover states are accurate for single and multi-channel pipetting steps

### CODE REVIEW

File: protocol-designer/src/components/DeckSetup/DeckSetup.js [hunk 1]
```diff
@@ -48,66 +51,79 @@ const DeckSetup = (props: Props) => {
...
+                  width={380}
+                  innerDivProps={{
+                    className: styles.deck_instructions,
+                  }}
                 >
                   {headerMessage}
-                </RobotCoordsText>
+                </RobotCoordsForeignDiv>
+
                 {map(slots, (slot, slotId) => {
```
Comment (IanLondon) [inline protocol-designer/src/components/DeckSetup/DeckSetup.js]: Flow isn't inferring this type so the block is untyped, could you do ` {map(slots, (slot: $Values<typeof slots>, slotId) => {` or something like that?

File: protocol-designer/src/components/DeckSetup/DeckSetup.js [hunk 2]
```diff
@@ -4,19 +4,22 @@ import map from 'lodash/map'
 import filter from 'lodash/filter'
```
Comment (IanLondon) [inline protocol-designer/src/components/DeckSetup/DeckSetup.js]: `DeckSetup` component and container no longer needs `ingredSelectionMode` prop :bomb:

File: protocol-designer/src/components/modals/FilePipettesModal/PipetteFields.js [hunk 3]
```diff
@@ -31,6 +23,25 @@ type Props = {
 
 export default function ChangePipetteFields(props: Props) {
   const { values, handleChange } = props
+
+  const tiprackOptions = useMemo(() => {
+    const allDefs = getAllDefinitions()
+    return reduce(
+      allDefs,
+      (acc, def) => {
```
Comment (IanLondon) [inline protocol-designer/src/components/modals/FilePipettesModal/PipetteFields.js]: could you do `(acc, def: $Values<typeof allDefs>) => {`? Sadly Flow doesn't infer iterands in reduce/map :crying_cat_face:

File: protocol-designer/src/localization/en/deck.json [hunk 4]
```diff
@@ -15,7 +15,9 @@
       "delete": "Delete"
     },
     "slot": {
-      "add_labware": "Add Labware"
+      "add_labware": "Add Labware",
+      "drag_to_new_slot": "Drag To New Slot",
+      "place_here": "Place Here"
```
Comment (IanLondon) [inline protocol-designer/src/localization/en/deck.json]: "View Liquids" is still hard-coded in `protocol-designer/src/components/DeckSetup/LabwareOverlays/BrowseLabware.js`, could go here

File: components/src/deck/RobotWorkSpace.js [hunk 5]
```diff
@@ -35,11 +50,12 @@ function RobotWorkSpace(props: Props) {
     <svg
       className={styles.robot_work_space}
       viewBox={viewBox || wholeDeckViewBox}
+      ref={el => (wrapperRef.current = el)}
```
[the remark this answers is not in the document]
Comment (b-cooper) [reply components/src/deck/RobotWorkSpace.js]: yeah, it's left over from hopeful flow svg ref finagling

File: protocol-designer/src/components/DeckSetup/DeckSetup.js [hunk 6]
```diff
@@ -48,66 +51,79 @@ const DeckSetup = (props: Props) => {
...
               'doorStops',
               'metalFrame',
-              'removalHandles',
+              'removalHandle',
               'screwHoles',
             ]}
             deckDef={deckDef}
-            viewBox={`-10 -10 ${410} ${390}`}
+            viewBox={`-10 -10 ${410} ${390}`} // viewbox for small
+            // viewBox={`-10 -10 ${460} ${452}`} // viewbox for mid
```
[the remark this answers is not in the document]
Comment (b-cooper) [reply protocol-designer/src/components/DeckSetup/DeckSetup.js]: temporary styling for design demos, these magic numbers will be removed in the next PR (with the approved design)

File: components/src/deck/RobotWorkSpace.js [hunk 7]
```diff
@@ -1,21 +1,36 @@
...
+    y: number
+  ): { x: number, y: number } => {
+    if (!wrapperRef.current) return { x: 0, y: 0 }
+    const cursorPoint = wrapperRef.current.createSVGPoint()
+
+    cursorPoint.x = x
+    cursorPoint.y = y
+
+    return cursorPoint.matrixTransform(
+      wrapperRef.current.getScreenCTM().inverse()
```
[the remark this answers is not in the document]
Comment (b-cooper) [reply components/src/deck/RobotWorkSpace.js]: I'll throw a note in there. Good catch

**Pull request merged**